integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -1088,3 +1073,81 @@ def do_it(spark):
         df = two_col_df(spark, StringGen('k{1,5}'), ArrayGen(MapGen(StringGen('a{1,5}', nullable=False), StringGen('[ab]{1,5}'))))
         return df.groupBy('a').agg(f.min(df.b[1]["a"]))
     assert_gpu_and_cpu_are_equal_collect(do_it)
+
+
+@ignore_order
+@allow_non_gpu('HashAggregateExec', 'Alias', 'AggregateExpression', 'Cast',
+  'HashPartitioning', 'ShuffleExchangeExec', 'Average')
+@pytest.mark.parametrize('data_gen', [_grpkey_strings_with_extra_nulls], ids=idfn)
+@pytest.mark.parametrize('conf', get_params(_confs, params_markers_for_confs), ids=idfn)
+@pytest.mark.parametrize('ansi_enabled', ['true', 'false'])
+def test_hash_grpby_avg_nulls_ansi(data_gen, conf, ansi_enabled):
+    local_conf = copy_and_update(conf, {'spark.sql.ansi.enabled': ansi_enabled})
+    assert_gpu_and_cpu_are_equal_collect(
```

Comment:
I m confused why want to combine this into a single test. Doing it this way makes it so we cannot verify that we did fallback for Average and it also means that we explicitly allow things to not be on the GPU when we know that they should be in non-ansi mode.

Reply:
Sorry, I moved that test down together with the other ANSI stuff, I didn't change it. Will split it, totally agree.

Reply:
Done here: https://github.com/NVIDIA/spark-rapids/pull/3597/commits/4ddbb6bdfbbe92a3fee286a022c321ebe33d162a